src/NzbDrone.Core/Notifications/MailGun/MailGunSettings.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using System.Net;
+using System.Runtime.CompilerServices;
+using FluentValidation;
+using NzbDrone.Core.Annotations;
+using NzbDrone.Core.ThingiProvider;
+using NzbDrone.Core.Validation;
+
+namespace NzbDrone.Core.Notifications.MailGun
+{
+    public class MailGunSettingsValidator : AbstractValidator<MailGunSettings>
+    {
+        public MailGunSettingsValidator()
+        {
+            RuleFor(c => c.ApiKey).NotEmpty();
+            RuleFor(c => c.From).NotEmpty();
+            RuleFor(c => c.Recipients).NotEmpty();
+        }
+    }
+    
+    public class MailGunSettings : IProviderConfig
+    {
+      private static readonly  MailGunSettingsValidator Validator = new MailGunSettingsValidator();
+
+      public MailGunSettings()
+      {
+          BaseUrlUs = "https://api.mailgun.net/v3";
+          BaseUrlEu = "https://api.eu.mailgun.net/v3";
+          Recipients = new string[] { };
+      }
+      
+      public string BaseUrlUs { get; set; }
+      public string BaseUrlEu { get; set; }
+      
+      [FieldDefinition(1, Label = "API Key", HelpText = "The API key generated from MailGun")]
+      public string ApiKey { get; set; }
+      
+      [FieldDefinition(2, Label = "Use EU endpoint?", HelpText = "You can choose to use the EU MailGun endpoint with this", Type = FieldType.Checkbox)]
+      public bool IsEu { get; set; }
```

Comment:
`UseEuEndpoint` is a better name for this prop (just like the label) as it has nothing to do with the source of the request, but the destination.